package.json

```diff
@@ -55,7 +55,7 @@
     "test": "npm run test:ts && npm run jest",
     "test:deploy": "test/deploy/validate-package.sh",
     "test:ts": "tsc -p test --noEmit",
-    "test:integration": "jest --coverageReporters text-summary -- test/integration",
+    "test:integration": "jest --coverageReporters --forceExit text-summary -- test/integration",
```

Comment:
this change needs some explanation and reasoning.

At first, the RedisResourceLockerIntegration tests would hang in CI.
Jest suggested this was an open promise that was and/or is not handled by the time the tests complete.

Running Jest with `--runInBand --detectOpenHandles` should tell you where the promise leak is. Using these flags didn't provide any extra output or info (not an unusual issue looking at old issues no the jest Github repo).

Looking into it some more i found people using databases having the same issue(in the examples i found they could simply call close() on whatever db they were using).
I found online that node processes are not able to quit as long as it has an open connection / socket.
I am 99% sure that this is the issue, the connection with the Redis server doesn't close.

Redlock wants you to call `.quit()` on the Redlock instance to correctly close down all connections to a Redis server. We don't have control over the Redlock instance in the integration tests, so this is not an option.

Digging into the Jest docs i found that the flag `--forceExit` should be used when you can't do a proper teardown. Like in this situation, i cannot just call `quit()` on the Redlock instance.

Reply:
Would this be solved by adding a public `quit` function to `RedisResourceLocker` that handles closing the connection, and calling that in the integration test? Not sure yet which solution I would prefer but want to know first if this is an option.

Reply:
I don't think a public quit() function is a good idea and I am not sure how we would get the RedisResourceLocker's instance in the tests. Maybe with some componentsjs magic that i don't know of.

Reply:
You can get the instance similar to how you get the initializer and handler now: would require an extra entry in the `urn:solid-server:test:Instances` object. It could be an idea to create a new `Destructor` interface that has a `quit` function for classes that need to be called when the server is shutting down if we want a clean exit. But that might be too much overhead instead of just quitting the process. @RubenVerborgh do you have an opinion on this?

Reply:
> `--forceExit`

This would definitely bite us later.

We need to find a way to do a good teardown. The public method is fine with me.